examples/data_preprocessing/[WIP]_chexpert_dataset/Prepare_CheXpert_data.ipynb

```diff
@@ -0,0 +1,478 @@
+    "        \n",
+    "        if self.transform_sequence is not None:\n",
+    "            patient_image = self.transform_sequence(patient_image) #apply the transform_sequence if one is specified\n",
+    "        \n",
+    "        else:\n",
+    "            #even if no other transformation is applied, the image should be turned into a tensor\n",
+    "            to_tensor = transforms.ToTensor()\n",
+    "            patient_image = to_tensor(patient_image)\n",
+    "        \n",
+    "        return patient_image, torch.tensor(image_labels)\n",
+    "    \n",
+    "    def __len__(self):\n",
+    "        return self.number_of_images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "chexpert_train = prepare_CheXpert(path, transform_sequence = transform_list)\n",
+    "chexpert_valid = prepare_CheXpert(path, subset = \"valid\", transform_sequence = transform_list)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For a given index, the getitem function returns the tensor representing the preprocessed image as well as a tensor containing all of the labels for this observation.\n",
+    "Let's take a look at what getitem returns for the training example with index 0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "chexpert_train.__getitem__(0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Shape of image tensor: torch.Size([3, 320, 320])\n",
+      "Shape of label tensor: torch.Size([12])\n"
+     ]
+    }
+   ],
+   "source": [
+    "shape_of_image_tensor = chexpert_train.__getitem__(0)[0].shape\n",
+    "shape_of_label_tensor = chexpert_train.__getitem__(0)[1].shape\n",
+    "\n",
+    "print(\"Shape of image tensor:\", shape_of_image_tensor)\n",
+    "print(\"Shape of label tensor:\", shape_of_label_tensor)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Storing the preprocessed data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The tensors representing the preprocessed images and the tensors containing the labels can be stored on your computer using joblib. If you want to do this, you can run the following code but please be aware that the resulting joblib files might become very large, especially for the training data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 234/234 [00:01<00:00, 180.44it/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "filename = 'chexpert_data_train.joblib'\n",
```

Comment:
the filenames should rather be defined at the begging of the tutorial (e.g. in the cell 2)

Reply:
I will change that, once I have the section for storing the data finalized.